### DISCUSSION TITLE
What does "unlock_failed" mean in the debugging reflections

### DISCUSSION THREAD

**Discussion by blairio** [mhenrixon/sidekiq-unique-jobs#773] [Q&A] [answered]
The documentation https://github.com/mhenrixon/sidekiq-unique-jobs#unlock_failed says:
> This is not got, this is worth

I don't understand what that means, and would really like to know more about the is reflection.

We ended up using this to find jobs that were scheduled before an upgrade that got dropped in after the upgrade because the lock arguments changed.

But since the upgrade, we are still seeing these messages in the logs occasionally, and I am worried that jobs are being dropped regularly.

I have seen this on jobs using until_and_while_executing and until_executed

Versions:
- gem 'sidekiq-pro', '5.5.8'
- gem 'sidekiq-scheduler', '4.0.3'
- gem 'sidekiq-unique-jobs', '7.1.29'
- gem 'redis', '4.8.0'
- gem 'rails', '5.2.7'

Any insight would be appreciated,
Thanks,
=Blair.

**Comment by mhenrixon** [comment, marked as answer]
https://github.com/mhenrixon/sidekiq-unique-jobs/pull/776

**Reply by blairio** [reply]
Thanks for the update. That sounds bad. Does that mean the job didn't run, and may not ever run even if a new one is enqueued? Is there a recommended mitigation when this happens?

**Reply by mhenrixon** [reply]
Usually, it is rare. The reaper should clean that up for you.

**Reply by mhenrixon** [reply]
The job that reported it was likely completed but any new jobs with the same argument wouldn't be able to run until the reaper cleaned up the orphaned lock.
